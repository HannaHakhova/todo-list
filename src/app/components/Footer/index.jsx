import React, {Component} from 'react';
import {Menu, Button, Icon, Label} from 'semantic-ui-react';
import {arrayOf, func, shape, string, bool} from 'prop-types';

export default class Footer extends Component {
    constructor(props) {
        super(props);
        this.state = {activeItem: 'all'};
    }

    handleItemClick = (e, {name}) => {
        this.props.handleActiveItemChange(name);
        this.setState({activeItem: name});
    };

    render() {
        const {activeItem} = this.state;

        return (
            <Menu pointing secondary >
                <Menu.Item as='label' name='all' active={activeItem === 'all'} onClick={this.handleItemClick}>
                    <Icon name='list layout' /> All
                    <Label color='orange'>{this.props.todos.length}</Label>
                </Menu.Item>
                <Menu.Item as='label' name='completed' active={activeItem === 'completed'} onClick={this.handleItemClick}>
                    <Icon name='check' /> Completed
                    <Label color='teal' >{this.props.todos.filter(todo => todo.completed).length}</Label>
                </Menu.Item>
                <Menu.Item as='label' name='pending' active={activeItem === 'pending'} onClick={this.handleItemClick}>
                    <Icon name='hourglass end' /> Pending
                    <Label color='red' >{this.props.todos.filter(todo => !todo.completed).length}</Label>
                </Menu.Item>
                <Menu.Item>
                    <Button color='teal' onClick={this.props.handleClearCompleted} >
                        <Icon name='trash' /> Clear completed
                    </Button>
                </Menu.Item>
            </Menu>);
    }
}

Footer.propTypes = {
    todos: arrayOf(shape({
        title: string.isRequired,
        completed: bool.isRequired
    }).isRequired).isRequired,
    handleClearCompleted: func.isRequired,
    handleActiveItemChange: func.isRequired
};