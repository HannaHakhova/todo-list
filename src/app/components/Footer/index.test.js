/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow, mount} from 'enzyme';
import Footer from './index';

const component = shallow(<Footer todos={[{title: 'Test1', completed: false}]} handleClearCompleted={() => {}} handleActiveItemChange={() => {}} />);

describe('Footer', () => {
    it('matchs expected snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('calls handleItemClick when click on menu item', () => {
        const handleActiveItemChange = jest.fn();
        const wrapper = mount(<Footer todos={[]} handleClearCompleted={() => {}} handleActiveItemChange={handleActiveItemChange} />);
        wrapper.setState({activeItem: 'bar'});

        wrapper.find('MenuItem').first().simulate('click');

        expect(handleActiveItemChange).toBeCalledWith('all');
        expect(wrapper.state().activeItem).toEqual('all');
    });

    it('calls handleClearCompleted when click on the last menu item', () => {
        const handleClearCompleted = jest.fn();
        const wrapper = shallow(<Footer todos={[]} handleClearCompleted={handleClearCompleted} handleActiveItemChange={() => {}} />);

        wrapper.find('Button').simulate('click');

        expect(handleClearCompleted).toBeCalled();
    });
});