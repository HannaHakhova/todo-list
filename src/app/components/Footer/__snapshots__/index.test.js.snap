// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matchs expected snapshot 1`] = `
<Menu
  pointing={true}
  secondary={true}
>
  <MenuItem
    active={true}
    as="label"
    name="all"
    onClick={[Function]}
  >
    <Icon
      as="i"
      name="list layout"
    />
     All
    <Label
      color="orange"
    >
      1
    </Label>
  </MenuItem>
  <MenuItem
    active={false}
    as="label"
    name="completed"
    onClick={[Function]}
  >
    <Icon
      as="i"
      name="check"
    />
     Completed
    <Label
      color="teal"
    >
      0
    </Label>
  </MenuItem>
  <MenuItem
    active={false}
    as="label"
    name="pending"
    onClick={[Function]}
  >
    <Icon
      as="i"
      name="hourglass end"
    />
     Pending
    <Label
      color="red"
    >
      1
    </Label>
  </MenuItem>
  <MenuItem>
    <Button
      as="button"
      color="teal"
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="trash"
      />
       Clear completed
    </Button>
  </MenuItem>
</Menu>
`;
